thrift0.13/src/main/java/com/linecorp/armeria/common/thrift/ThriftProtocolFactoryProvider.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.thrift;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Set;
+
+import org.apache.thrift.protocol.TProtocolFactory;
+
+import com.linecorp.armeria.common.SerializationFormat;
+
+/**
+ * SPI Provider for links from {@link SerializationFormat} to {@link TProtocolFactory}.
+ */
+@FunctionalInterface
+public interface ThriftProtocolFactoryProvider {
+    /**
+     * Pair of {@link SerializationFormat} and {@link TProtocolFactory}.
+     */
+    final class ThriftSerializationFormat {
+        private final SerializationFormat serializationFormat;
+        private final TProtocolFactory tProtocolFactory;
+
+        public ThriftSerializationFormat(SerializationFormat serializationFormat,
+                                         TProtocolFactory tProtocolFactory) {
+            this.serializationFormat = requireNonNull(serializationFormat, "serializationFormat");
+            this.tProtocolFactory = requireNonNull(tProtocolFactory, "tProtocolFactory");
+        }
+
+        public SerializationFormat getSerializationFormat() {
+            return serializationFormat;
+        }
+
+        public TProtocolFactory getTProtocolFactory() {
+            return tProtocolFactory;
+        }
+    }
+
+    /**
+     * Accessed configured {@link ThriftSerializationFormat}s for this SPI provider.
+     *
+     * @return an immutable view
+     */
+    Set<ThriftSerializationFormat> thriftSerializationFormats();
```

Comment:
How about renaming `ThriftSerializationFormat` to `Entry`, like we did in `SerializationFormatProvider`?